test/transcoding/SPV_KHR_integer_dot_product-nonsat.ll

```diff
@@ -0,0 +1,292 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text --spirv-ext=+SPV_KHR_integer_dot_product -o %t.txt
+; RUN: FileCheck < %t.txt %s --check-prefix=CHECK-SPIRV
+; RUN: llvm-spirv --spirv-ext=+SPV_KHR_integer_dot_product %t.bc -o %t.spv
+; RUN: llvm-spirv -r %t.spv -o %t.rev.bc
+; RUN: llvm-dis < %t.rev.bc | FileCheck %s --check-prefix=CHECK-LLVM
+
+; CHECK-SPIRV: Capability DotProductInputAllKHR
+; CHECK-SPIRV: Capability DotProductInput4x8BitKHR
+; CHECK-SPIRV: Capability DotProductInput4x8BitPackedKHR
+; CHECK-SPIRV: Capability DotProductKHR
+; CHECK-SPIRV: Extension "SPV_KHR_integer_dot_product"
+
+; CHECK-SPIRV-DAG: TypeInt [[#I8:]] 8
+; CHECK-SPIRV-DAG: TypeInt [[#I16:]] 16
+; CHECK-SPIRV-DAG: TypeInt [[#I32:]] 32
+; CHECK-SPIRV-DAG: TypeInt [[#I64:]] 64
+
+target datalayout = "e-p:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024"
+target triple = "spir-unknown-unknown"
+
+; Packed vector format: 32-bit scalar interpreted as v4i8.
+; CHECK-LLVM: @TestNonSatPacked
+; CHECK-SPIRV: Function
+
+; CHECK-LLVM: call spir_func i8 @_Z22__spirv_SDotKHR_Ruchariii
+; CHECK-LLVM: call spir_func i16 @_Z23__spirv_SDotKHR_Rushortiii
+; CHECK-LLVM: call spir_func i32 @_Z21__spirv_SDotKHR_Ruintiii
+; CHECK-LLVM: call spir_func i64 @_Z22__spirv_SDotKHR_Rulongiii
+
+; CHECK-LLVM: call spir_func i8 @_Z22__spirv_UDotKHR_Ruchariii
+; CHECK-LLVM: call spir_func i16 @_Z23__spirv_UDotKHR_Rushortiii
+; CHECK-LLVM: call spir_func i32 @_Z21__spirv_UDotKHR_Ruintiii
+; CHECK-LLVM: call spir_func i64 @_Z22__spirv_UDotKHR_Rulongiii
+
+; CHECK-LLVM: call spir_func i8 @_Z23__spirv_SUDotKHR_Ruchariii
+; CHECK-LLVM: call spir_func i16 @_Z24__spirv_SUDotKHR_Rushortiii
+; CHECK-LLVM: call spir_func i32 @_Z22__spirv_SUDotKHR_Ruintiii
+; CHECK-LLVM: call spir_func i64 @_Z23__spirv_SUDotKHR_Rulongiii
+
+; CHECK-SPIRV: 6 SDotKHR [[#I8]] [[#]] [[#]] [[#]] 0
```

Comment:
I suggest not to check number of words

Reply:
Any reason why not?  I thought it would be valuable to check the number of words, because these instructions take an optional "Packed Vector Format" operand; see http://htmlpreview.github.io/?https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/KHR/SPV_KHR_integer_dot_product.html

Reply:
Agree